torch_xla/csrc/torch_util.h

```diff
@@ -5,9 +5,33 @@
 #include <c10/util/Optional.h>
 
 #include "tensorflow/compiler/xla/shape.h"
+#include "torch/csrc/lazy/core/dynamic_ir.h"
 #include "torch/csrc/lazy/core/hash.h"
+#include "torch/csrc/lazy/core/tensor.h"
+#include "torch/csrc/lazy/core/util.h"
+
 namespace torch_xla {
 
+// Unpack SymInt objects into their building blocks
+struct SymIntElements {
+ public:
+  SymIntElements(c10::SymInt& size) { SetSymIntNodeElements(size); }
+
+  SymIntElements(c10::SymIntArrayRef& size) {
+    std::vector<c10::SymInt> _sizes = torch::lazy::ToVector<c10::SymInt>(size);
+    for (auto& _size : _sizes) {
+      SetSymIntNodeElements(_size);
+    }
+  }
+
+  std::vector<torch::lazy::NodePtr> size_nodes;
+  std::vector<int64_t> upper_bounds;
+  std::vector<bool> dynamic_dims;
```

Comment:
Can we renamed name with `_` at the end and move all of them to private then provide a getter(I don't think we need a setter).

Reply:
We likely need setters. I will push setters at a later stage when the use for them is evident.